(#504) Fix inability to share dead threads

K1rakishou opened the pull request:
Closes #504

K1rakishou committed:
(#504) Fix inability to share dead threads

K1rakishou committed:
(#504) Separate desktopUrl method into two methods: desktopUrlForThread and desktopUrlForPost

Adamantcheese commented on Kuroba/app/src/main/java/com/github/adamantcheese/chan/core/site/SiteUrlHandler.java:
```diff
@@ -16,21 +16,23 @@
+    abstract public boolean respondsTo(HttpUrl url);
 
-    boolean respondsTo(HttpUrl url);
+    public String desktopUrlForThread(Loadable loadable) {
```
I don't see a reason for having two methods here and changing the architecture so much. If you need the thread url, just set the post number to be the loadable number and have the various implementations deal with it. The only one that actually uses that postNo field is 4chan anyways.

K1rakishou replied:
Well yes, but I think it's more obvious. If you want to share a thread you do not need to pass a post no. And if you want to share a concrete post then you need post no.

K1rakishou committed:
(#504) Merge branch 'cheese-multi-feature' into (#504)-dead-threads-sharing

# Conflicts:
#	Kuroba/app/src/main/java/com/github/adamantcheese/chan/core/site/SiteUrlHandler.java
#	Kuroba/app/src/main/java/com/github/adamantcheese/chan/core/site/sites/Arisuchan.java
#	Kuroba/app/src/main/java/com/github/adamantcheese/chan/core/site/sites/Lainchan.java
#	Kuroba/app/src/main/java/com/github/adamantcheese/chan/core/site/sites/Sushichan.java
#	Kuroba/app/src/main/java/com/github/adamantcheese/chan/core/site/sites/Wired7.java
#	Kuroba/app/src/main/java/com/github/adamantcheese/chan/core/site/sites/dvach/Dvach.java

Adamantcheese commented:
Closed with https://github.com/Adamantcheese/Kuroba/commit/a324190efa70463152c752b28544f67c5c5fcfff.
